src/utils/i18n.tsx

```diff
@@ -1,39 +1,34 @@
 // Copyright (c) 2018 Ultimaker B.V.
-import { po } from 'gettext-parser';
 import * as React from 'react';
+import { po } from 'gettext-parser';
 
 import Gettext = require('node-gettext');
 
-// type for available languages
-export type Languages = 'en-US' | 'nl-NL';
-
-// interface for translation items that are passed into initialize
-export interface TranslationListItem {
-    name: Languages
-    source: any
-}
-
 /**
  * Assorted methods for translations.
  */
 export class I18n {
     private static _gt = new Gettext({ debug: false })
 
-    private static _defaultLanguage = 'en-DEV'
+    private static _defaultLanguage = 'en'
 
-    private static _supportedLanguages: Languages[] = ['en-US', 'nl-NL']
+    private static _supportedLanguages: string[] = []
 
     /**
-    * Initialize the Gettext context.
-    */
-    public static async initialize(translations: TranslationListItem[]) {
-        // load the translation files
-        translations.forEach(async translation => (
-            this._loadTranslation(translation.name, translation.source)
-        ));
+     * Initialize the Gettext context.
+     * @param translations The translation keys with files to load.
+     */
+    public static async initialize(translations: { [s: string]: string; }): Promise<void> {
+        // register supported languages
+        Object.keys(translations).forEach((language) => {
```

Comment:
Can't we use `Object.keys(translations).forEach(I18n._supportedLanguages.push)`?

Reply:
That would be cleaner indeed.

Reply:
nvm, the function signatures of forEach and push don't match, so typescript doesn't allow it.